### PR TITLE
UEFI boot order definition: set the device currently booted as the first entry

### LINKED ISSUE #338: aii-ks 21.12: regex for identifying PXE boot entries not strict enough
`aii-ks` has a great feature that allows to redefine the UEFI boot order to have the PXE entries first. Identifying the PXE entries relies on a regex as there is no other way to do it. But this regex is not strict enough on some machines, like HP servers, where the UEFI boot entry list contains things like `Boot Menu`, `Network Boot`, `PXE Boot`... that are just referring to LILO menus (clearly not what we want) and not to PXE entries. 

I attach the script 
[boot_pxe_first.sh](https://github.com/quattor/aii/files/11969767/boot_pxe_first.sh.txt) I use to reorder the boot list on an installed system where I used a slightly different regex that works at least on Dell and HP servers.

Michel

### PR DESCRIPTION
**Pull request by jouvin**
Fixes #338

### CODE REVIEW

File: aii-ks/src/main/perl/ks.pm [hunk 1]
```diff
@@ -1,4 +1,30 @@
-#${PMpre} NCM::Component::ks${PMpost}
```
Comment (stdweird) [inline aii-ks/src/main/perl/ks.pm:1]: please revert this part

File: aii-ks/src/main/perl/ks.pm [hunk 2]
```diff
@@ -17,7 +43,7 @@ our $EC = LC::Exception::Context->new->will_store_all;
 
 our $this_app = $main::this_app;
 # Modules that may be interesting for hooks.
-our @EXPORT_OK = qw (ksuserhooks ksinstall_rpm get_repos replace_repo_glob);
+our @EXPORT_OK = qw (ksuserhooks ksinstall_rpm);
```
Comment (stdweird) [inline aii-ks/src/main/perl/ks.pm:46]: why is this being removed? @jouvin is your PR based on latest master?

Comment (jouvin) [reply aii-ks/src/main/perl/ks.pm:46]: You are right, it based on 21.12.1, I'll fix this.

**Comment by jouvin** [comment]
Should be fixed, thanks @stdweird for noticing. 

@jrha I see that you don't plan to add it into the upcoming release. Please, could I ask you to reconsider this as the current version is broken and lead to painful operations on recent HP hardware... I think the approach suggested by @wpoely86 and implemented here is much safer. @stdweird what do you think?

**Review by jrha — APPROVED** [review]

**Pull request merged**